src/main/java/org/elasticsearch/common/Booleans.java

```diff
@@ -45,6 +44,7 @@ public static boolean parseBoolean(char[] text, int offset, int length, boolean
         if (length == 5) {
             return !(text[offset] == 'f' && text[offset + 1] == 'a' && text[offset + 2] == 'l' && text[offset + 3] == 's' && text[offset + 4] == 'e');
         }
+        // nocommit check for known "true" values and throw exc if it's not recognized
         return true;
```

Comment:
This method just made my head explode. If we are going to do this, can we at least code it cleaner?

Something like this:

    private static final CharArraySet WAYS_TO_SAY_FALSE = { };